moveit_core/robot_model/src/robot_model.cpp

```diff
@@ -349,6 +354,8 @@ void RobotModel::buildGroupStates(const srdf::Model& srdf_model)
                                    "but that joint is not part of group '%s'",
                           group_state.name_.c_str(), jt->first.c_str(), jmg->getName().c_str());
       }
+      if (!remaining_joints.empty())
```

Comment:
Additionally to throwing a warning, default values should be 0 or the state might be invalid. Alternatively, we could also make this an error and don't add the state at all.

Reply:
The group state is essentially a map: joint name -> value. Hence, it is fine to leave some values "undefined" (not listed in the map) as long as `setToDefaultState` starts from a fully-defined state.
This should be the case anyway as the there might be joints outside the JMG, which will definitely not be modified by  `setToDefaultState`. The fake controller manager takes care of this (now).